### PR TITLE
[insteon] Convert legacy threads to use scheduler service

### PR DESCRIPTION
**Pull request by jsetton**
Initially, I was going to revert the change part of #17804 to the `LegacyPort` class as the thread names were already normalized through the `setParamsAndStart` wrapper method (not sure why I wasn't on the reviewer list?).

However, it made more sense to just convert all the legacy threads to use the scheduler service instead, removing unnecessary singleton instances in the process. I did run a successful test in my environment of these changes.

### CODE REVIEW

**Comment by jsetton** [comment]
@lsiepel hopefully, this PR can make it into the 4.3 release.

**Comment by lsiepel** [comment]
> (not sure why I wasn't on the reviewer list?).

@jlaur  do we need to add @jsetton to some group? I thought he would automatically be listed for review as he is in the codeowner file: https://github.com/openhab/openhab-addons/blob/7fc3d3b6857491af7411cd53e49c0948bdbf3d3a/CODEOWNERS#L173

> hopefully, this PR can make it into the 4.3 release.

As we have entered code freeze and the binding is not broken, it is better to merge after tomorow so it can be part of 4.3.1 as patch release.

**Comment by jlaur** [comment]
> @jlaur do we need to add @jsetton to some group?

We need to ask @kaikreuzer to add @jsetton to the contributor team. @kaikreuzer - since @jsetton has rewritten the binding and is now the binding maintainer, can you add him? 🙂

**Review by lsiepel — APPROVED** [review]
Thanks, LGTM

**Comment by jsetton** [comment]
@lsiepel can we make sure that this PR gets added to 4.3.1 release as well? Thanks.

**Comment by lsiepel** [comment]
> @lsiepel can we make sure that this PR gets added to 4.3.1 release as well? Thanks.

Why is that important? As it does not actually fix a bug?

**Comment by jsetton** [comment]
>  As it does not actually fix a bug?

This PR is labelled as such. It's technically reverting an unnecessary change that is currently part of the 4.3 release. You actually mentioned that it would be part of 4.3.1 above.

**Comment by lsiepel** [comment]
> > As it does not actually fix a bug?
> 
> This PR is labelled as such. It's technically reverting an unnecessary change that is currently part of the 4.3 release. You actually mentioned that it would be part of 4.3.1 above.

Not exactly, i said _can be_, that is not the same. The label is there for the release notes, one could argue if it should be removed or not. But all that does not really  matter, as long as it is not merged/patched there is room for open discussion. I do see the benefit of this PR as it improves the code, that is why it is merged and available in the main branch.  But what do we gain (for users) when this is available in the next patch release?

IMHO when balancing risks vs gains, i'm at this moment not yet convinced this needs a backport.

**Comment by jsetton** [comment]
I leave it up to you. Based on the information I provided, I think it should be in 4.3.1 as 5.0 is 6 months away. That was the reason for asking if it could make it into 4.3. If I had visibility when the relevant PR was submitted, I could have corrected before the freeze.

**Comment by jsetton** [comment]
> We need to ask @kaikreuzer to add @jsetton to the contributor team. @kaikreuzer - since @jsetton has rewritten the binding and is now the binding maintainer, can you add him? 🙂

Any chance that I can be added to the contributor team? Thanks.

**Comment by kaikreuzer** [comment]
@jsetton Most certainly! Sorry, I missed the previous message from @jlaur.
I've added you now!

**Pull request merged**
